File: Day-51/ArraySum.py
def ArraySum(arrOne, arrTwo):
    i = len(arrOne)-1
    j = len(arrTwo)-1
    
    carry = 0
    returnArray = []
    while(i>=0 and j>=0):
        sum = arrOne[i] + arrTwo[j] + carry
        carry = sum//10
        sum = sum % 10
        i -= 1
        j -= 1
        returnArray.append(int(sum))
    
    while(i>=0):
        sum = arrOne[i] + carry
        carry = sum//10
        sum = sum % 10
        i-=1
        returnArray.append(int(sum))
    
    while(j>=0):
        sum = arrTwo[j] + carry
        carry = sum//10
        sum = sum % 10
        i-=1
        returnArray.append(int(sum))
    
    if(carry>0):
        returnArray.append(carry)
    returnArray.reverse()
    return returnArray


print(ArraySum([4, 5, 1], [7, 4, 5]))

File: Day-51/ArraySum.js
function ArraySum(arr1, arr2) {
  let i = arr1.length - 1;
  let j = arr2.length - 1;
  let carry = 0;

  let returnArray = [];

  while (i >= 0 && j >= 0) {
    let sum = arr1[i] + arr2[j] + carry;
    carry = sum / 10;
    sum = sum % 10;
    returnArray.unshift(parseInt(sum));
    i--;
    j--;
  }

  while (i >= 0) {
    let sum = arr1[i] + carry;
    carry = sum / 10;
    sum = sum % 10;
    returnArray.unshift(parseInt(sum));
    i--;
  }
  while (j >= 0) {
    let sum = arr2[j] + carry;
    carry = sum / 10;
    sum = sum % 10;
    returnArray.unshift(parseInt(sum));
    j--;
  }

  if (carry > 0) {
    returnArray.unshift(parseInt(carry));
  }

  return returnArray;
}

console.log(ArraySum([4, 5, 1], [7, 4, 5]));
